src/sparseml/keras/optim/mask_pruning.py

```diff
@@ -51,6 +59,54 @@ def target_sparsity(self, step: int, **kwargs) -> float:
         """
         raise NotImplementedError("Not implemented")
 
+    @abc.abstractmethod
+    def get_config(self):
+        raise NotImplementedError("Not implemented")
+
+
+class SchedulerRegistry:
```

Comment:
We should be able to get rid of this class entirely and avoid the potential confusion that comes with a more complicated setup. Specifically, from_config would be a class method on the PruningScheduler and an internal registry of the derived classes can be created by overriding the __new__ definition

Reply:
Agreed that decorator is not a good solution here for registering classes. I've removed it and fixed using __init_subclass__.